packages/devtools/lib/src/service.dart

```diff
@@ -6,13 +6,51 @@ import 'dart:async';
 import 'dart:html' hide Event;
 import 'dart:typed_data';
 
+import 'package:sse/client/sse_client.dart';
+import 'package:vm_service/utils.dart';
+
 import 'vm_service_wrapper.dart';
 
-Future<VmServiceWrapper> connect(Uri uri, Completer<Null> finishedCompleter) {
-  final WebSocket ws = WebSocket(uri.toString());
+void _connectWithSse(
+  Uri uri,
+  Completer<VmServiceWrapper> connectedCompleter,
+  Completer<Null> finishedCompleter,
+) {
+  uri = uri.scheme == 'sse'
+      ? uri.replace(scheme: 'http')
+      : uri.replace(scheme: 'https');
+  final SseClient client = SseClient('$uri');
+  client.onOpen.listen((_) {
+    final Stream<String> stream = client.stream.asBroadcastStream();
+    stream.listen((_) {}, onError: (error) {
+      if (!connectedCompleter.isCompleted) {
+        connectedCompleter.completeError(error);
+      }
+    });
 
-  final Completer<VmServiceWrapper> connectedCompleter =
-      Completer<VmServiceWrapper>();
+    final VmServiceWrapper service = VmServiceWrapper.fromNewVmService(
+      stream,
+      (String message) => client.sink.add(message),
+    );
+
+    client.sink.done.then((_) {
```

Comment:
How about `whenComplete`? instead of `.then`?

Reply:
TIL that exists.